hibernate-core/src/test/java/org/hibernate/orm/test/embeddable/EmbeddableAscDescQueryTest.java

```diff
@@ -0,0 +1,372 @@
+						);
+						entityAS = session.createQuery( query ).getResultList();
+						assertThat( entityAS ).hasSize( 2 );
+
+						entityA1 = entityAS.get( 1 );
+						assertThat( entityA1.getId() ).isEqualTo( 1 );
+						assertThat( entityA1.getComponentB() ).isNotNull();
+						assertThat( entityA1.getComponentB().getEntityD() ).isNotNull();
+						assertThat( entityA1.getComponentB().getEntityC() ).isNull();
+
+					}
+					finally {
+						session.getTransaction().rollback();
+					}
+				}
+		);
+	}
+
+	@Test
+	public void testQueryAsc(SessionFactoryScope scope) {
+		scope.inTransaction(
+				session -> {
+					CriteriaBuilder cb = session.getCriteriaBuilder();
+					CriteriaQuery<EntityA> query = cb.createQuery( EntityA.class );
+					Root<EntityA> root = query.from( EntityA.class );
+					query.select( root ).orderBy(
+							cb.asc( root.get( "id" ) )
+					);
+					List<EntityA> entityAS = session.createQuery( query ).getResultList();
+					assertThat( entityAS ).hasSize( 2 );
+
+					EntityA entityA1 = entityAS.get( 0 );
+					assertThat( entityA1.getId() ).isEqualTo( 1 );
+					assertThat( entityA1.getComponentB() ).isNotNull();
+					assertThat( entityA1.getComponentB().getEntityD() ).isNotNull();
+					assertThat( entityA1.getComponentB().getEntityC() ).isNull();
+
+				}
+		);
+	}
+
+	@Test
+	public void testQueryDesc(SessionFactoryScope scope) {
+		scope.inTransaction(
+				session -> {
+					CriteriaBuilder cb = session.getCriteriaBuilder();
+					CriteriaQuery<EntityA> query = cb.createQuery( EntityA.class );
+					Root<EntityA> root = query.from( EntityA.class );
+					query.select( root ).orderBy(
+							cb.desc( root.get( "id" ) )
+					);
+					List<EntityA> entityAS = session.createQuery( query ).getResultList();
+					assertThat( entityAS ).hasSize( 2 );
+
+					EntityA entityA1 = entityAS.get( 1 );
+					assertThat( entityA1.getId() ).isEqualTo( 1 );
+					assertThat( entityA1.getComponentB() ).isNotNull();
+					assertThat( entityA1.getComponentB().getEntityD() ).isNotNull();
+					assertThat( entityA1.getComponentB().getEntityC() ).isNull();
+				}
+		);
+	}
+
+
+	@Test
+	public void testQueryEntityB(SessionFactoryScope scope) {
+		scope.inTransaction(
+				session -> {
+					List<EntityB> entityBS = session.createQuery( "select b from EntityB b" ).getResultList();
+					assertThat( entityBS ).hasSize( 1 );
+
+					EntityB entrityB = entityBS.get( 0 );
+					EntityA entityA = entrityB.getEntityA();
+					assertThat( entityA ).isNotNull();
+					assertThat( entityA.getId() ).isEqualTo( 1 );
+					assertThat( entityA.getComponentB() ).isNotNull();
+					assertThat( entityA.getComponentB().getEntityD() ).isNotNull();
+					assertThat( entityA.getComponentB().getEntityC() ).isNull();
+				}
+		);
+	}
+
+	@Entity(name = "EntityA")
+	@Table(name = "t_entity_a")
+	public static class EntityA {
+
+		@Id
+		private Integer id;
+
+		private String name;
+
+		@Embedded
+		private ComponentA componentA;
```

Comment:
This component contains a self-reference association to `EntityA`, and is causing constraints errors on MySQL and MariaDB when executing `delete from EntityA` mutation query. I think we should clear the association before deleting all rows in the test's `tearDown()` method.

Reply:
something like 
```
session.createMutationQuery( "delete from EntityB" ).executeUpdate();
session.createMutationQuery( "delete from EntityC" ).executeUpdate();
List<EntityA> entityAs = session.createQuery( "select a from EntityA a", EntityA.class ).list();
entityAs.forEach(
	entityA -> {
		ComponentA componentA = entityA.getComponentA();
		if ( componentA != null ) {
			componentA.setEntityA( null );
			session.flush();
		}
				session.remove( entityA );
	}
);
session.createMutationQuery( "delete from EntityD" ).executeUpdate();
				
```
?